src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -292,28 +334,58 @@ public AbstractAnalyzer.AnalyzerFactory getQueryAnalyzerFactory()
         return queryAnalyzerFactory;
     }
 
-    public boolean isIndexed()
+    public View getView()
     {
-        return config != null;
+        return viewManager.getView();
+    }
+
+    /**
+     * @return total number of per-index open files
+     */
+    public int openPerIndexFiles()
+    {
+        return viewManager.getView().size() * Version.LATEST.onDiskFormat().openFilesPerIndex(this);
+    }
+
+    public void drop(Collection<SSTableReader> sstablesToRebuild)
+    {
+        viewManager.drop(sstablesToRebuild);
+    }
+
+    public boolean isNotIndexed()
+    {
+        return config == null;
     }
 
     /**
      * Called when index is dropped. Clear all live in-memory indexes and close
-     * analyzer factories.
+     * analyzer factories. Mark all {@link SSTableIndex} as released and per-column index files
+     * will be removed when in-flight queries completed and {@code obsolete} is true.
+     *
+     * @param obsolete true if index files should be deleted after invalidate; false otherwise.
      */
-    public void invalidate()
+    public void invalidate(boolean obsolete)
```

Comment:
This is never called with a false `obsolete`.

Reply:
Yes, this flag was used in a code path that doesn't exist here.